MekHQ/src/mekhq/gui/adapter/UnitTableMouseAdapter.java

```diff
@@ -366,7 +368,13 @@ public void actionPerformed(ActionEvent action) {
                             continue;
                         }
 
-                        overrideSkills(gui.getCampaign(), person, person.getPrimaryRole(), skillLevel.ordinal());
+                        Campaign campaign = gui.getCampaign();
+                        RandomSkillPreferences randomSkillPreferences = campaign.getRandomSkillPreferences();
+                        boolean useExtraRandomness = randomSkillPreferences.randomizeSkill();
+
+                        // We don't care about admin settings, as we're not going to have an admin here
+                        overrideSkills(false, false, useExtraRandomness,
+                              person, person.getPrimaryRole(), skillLevel);
```

Comment:
change the API of this function to accept the enum, or overload it to accept skill level instead of doing this thing of using ordinal instead of enum.